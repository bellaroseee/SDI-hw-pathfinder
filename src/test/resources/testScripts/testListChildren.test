# Test listChildren empty (no children)

CreateGraph graph1

AddNode graph1 A

ListChildren graph1 A

# Test listChildren sorted by weight

AddNode graph1 B
AddNode graph1 C
AddNode graph1 D

AddEdge graph1 A A 1.0
AddEdge graph1 A B 1.0
AddEdge graph1 A C 2.0
AddEdge graph1 A D 3.0
AddEdge graph1 A A 10.0
AddEdge graph1 A B 10.0

ListChildren graph1 A

# Test listChildren cyclic graph

AddEdge graph1 B A 1.0

ListChildren graph1 A