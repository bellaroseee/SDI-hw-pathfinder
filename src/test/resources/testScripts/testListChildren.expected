# Test listChildren empty (no children)

created graph graph1

added node A to graph1

the children of A in graph1 are:

# Test listChildren sorted by weight

added node B to graph1
added node C to graph1
added node D to graph1

added edge 1.000 from A to A in graph1
added edge 1.000 from A to B in graph1
added edge 2.000 from A to C in graph1
added edge 3.000 from A to D in graph1
added edge 10.000 from A to A in graph1
added edge 10.000 from A to B in graph1

the children of A in graph1 are: A(1.0) A(10.0) B(1.0) B(10.0) C(2.0) D(3.0)

# Test listChildren cyclic graph

added edge 1.000 from B to A in graph1

the children of A in graph1 are: A(1.0) A(10.0) B(1.0) B(10.0) C(2.0) D(3.0)